### PR TITLE
Fix Bug 890519 - Prevent module bailing earlier than expected for new we...

### PR DESCRIPTION
**Pull request by mjschranz**
...bmaker accounts

https://bugzilla.mozilla.org/show_bug.cgi?id=890519

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -64,6 +64,11 @@ module.exports = function ( app, options ) {
         uri: webmakerUrl + "user/" + id,
         json: true
       }, function ( error, response, body ) {
+        // User account wasn't found. Treat differently
```
Comment (Pomax) [inline index.js:67]: not so much "differently" - if 404, `this user is unknown, which triggers the "sign up new user" code paths`.

File: index.js [hunk 2]
```diff
@@ -64,6 +64,11 @@ module.exports = function ( app, options ) {
...
+        if ( response.statusCode === 404 ) {
+          return callback();
```
Comment (Pomax) [inline index.js:69]: does callback need arguments?

File: index.js [hunk 3]
```diff
@@ -164,8 +169,7 @@ module.exports = function ( app, options ) {
 
       res.json( { status: "okay" } );
     },
-    audience: options.audience,
-    middleware: options.middleware
```
Comment (Pomax) [inline index.js:168]: what role did middleware play before? does it need removing in this ticket? (I don't see code that is affected by this removal, so I'd rather not chance an accidental dependency-removal to fix a completely different issue. Good for a quick-ticket new issue)

File: index.js (hunk 1 above)
Comment (humphd) [inline index.js:67]: Because the order matters here (i.e., processing the `statusCode` _before_ `error`), let's add one more line to the comment to indicate that this has to happen first.

Because of this, which I think is OK as a band-aid but troubling long term, I'd suggest we file a follow-up to fix what we actually return from the login server so errors can be processed first.

File: index.js (hunk 3 above)
Comment (humphd) [inline index.js:168]: Why include this change here?  Seems unrelated.

File: index.js (hunk 2 above)
Comment (cadecairos) [inline index.js:69]: maybe we should do callback( null, null ) explicitly.

Comment (Pomax) [inline index.js:69]: if this follows node convention of `callback(err, result)`, then these arguments probably shouldn't both be null. `callback(null, { ... })` where the object's content indicated user creation will happen would be better (both for us, as well as new devs joining and being put on this code)

File: index.js (hunk 3 above)
Comment (mjschranz) [inline index.js:168]: Sorry, copy paste error. Good catch.

File: index.js (hunk 2 above)
Comment (mjschranz) [inline index.js:69]: Only issue there is that's breaking expected functionality elsewhere. I think `null, null` or just plain empty is good for now and can be solved in a follow up as requested by humph.
